### ISSUE TITLE
Fail to compile cpptest-lite on Linux with gcc

### ISSUE THREAD

**Issue by sadr0b0t** [doe300/cpptest-lite#3]
Hello, I'm trying to compile cpptest-lite on Linux,

do

~~~bash
 git clone https://github.com/doe300/cpptest-lite.git
 cd cpptest-lite/
 mkdir build
 cd build/
 cmake -DCPPTEST_LITE_CREATE_TESTS=ON -G "Unix Makefiles" ..
 cmake --build .
~~~

Get compilation errors, the first one:
~~~
TestSuite.h:210:30: error: expected ‘,’ before ‘...’ token
~~~

it poits here:
https://github.com/doe300/cpptest-lite/blob/master/include/TestSuite.h#L209
~~~cpp
            template<typename... T>
            TestMethod(const std::string& name, ParameterizedTestMethod<T...> method, const T... args) : name(name),
                functor([args..., method](Suite* suite) {(suite->*method)(args...);}), argString(joinStrings(args...))
            {
                   
            }
~~~

~~~bash
$ gcc --version
gcc (Ubuntu 4.8.4-2ubuntu1~14.04.1) 4.8.4
~~~

full error message:
~~~bash
$ cmake --build .
[  7%] Building CXX object CMakeFiles/cpptest-lite.dir/src/BDDSuite.cpp.o
In file included from /home/benderamp/Documents/1i7/local/cpptest-lite/include/cpptest.h:14:0,
                 from /home/benderamp/Documents/1i7/local/cpptest-lite/include/BDDSuite.h:11,
                 from /home/benderamp/Documents/1i7/local/cpptest-lite/src/BDDSuite.cpp:8:
/home/benderamp/Documents/1i7/local/cpptest-lite/include/TestSuite.h: In constructor ‘Test::Suite::TestMethod::TestMethod(const string&, Test::Suite::ParameterizedTestMethod<T ...>, const T ...)’:
/home/benderamp/Documents/1i7/local/cpptest-lite/include/TestSuite.h:210:30: error: expected ‘,’ before ‘...’ token
                 functor([args..., method](Suite* suite) {(suite->*method)(args...);}), argString(joinStrings(args...))
                              ^
/home/benderamp/Documents/1i7/local/cpptest-lite/include/TestSuite.h:210:30: error: expected identifier before ‘...’ token
/home/benderamp/Documents/1i7/local/cpptest-lite/include/TestSuite.h:210:33: error: expected ‘]’ before ‘,’ token
                 functor([args..., method](Suite* suite) {(suite->*method)(args...);}), argString(joinStrings(args...))
                                 ^
/home/benderamp/Documents/1i7/local/cpptest-lite/include/TestSuite.h:210:33: error: parameter packs not expanded with ‘...’:
/home/benderamp/Documents/1i7/local/cpptest-lite/include/TestSuite.h:210:33: note:         ‘args’
/home/benderamp/Documents/1i7/local/cpptest-lite/include/TestSuite.h: In lambda function:
/home/benderamp/Documents/1i7/local/cpptest-lite/include/TestSuite.h:210:33: error: expected ‘{’ before ‘,’ token
/home/benderamp/Documents/1i7/local/cpptest-lite/include/TestSuite.h: In constructor ‘Test::Suite::TestMethod::TestMethod(const string&, Test::Suite::ParameterizedTestMethod<T ...>, const T ...)’:
/home/benderamp/Documents/1i7/local/cpptest-lite/include/TestSuite.h:210:41: error: expected ‘)’ before ‘]’ token
                 functor([args..., method](Suite* suite) {(suite->*method)(args...);}), argString(joinStrings(args...))
                                         ^
/home/benderamp/Documents/1i7/local/cpptest-lite/include/TestSuite.h:210:41: error: expected ‘{’ before ‘]’ token
make[2]: *** [CMakeFiles/cpptest-lite.dir/src/BDDSuite.cpp.o] Ошибка 1
make[1]: *** [CMakeFiles/cpptest-lite.dir/all] Ошибка 2
make: *** [all] Ошибка 2
~~~

**Comment by sadr0b0t** [comment]
ok, I have replaced

~~~cpp
#ifdef __clang__
~~~
with

~~~cpp
#if defined(__clang__) || defined(__GNUC__)
~~~

here
https://github.com/doe300/cpptest-lite/blob/master/include/TestSuite.h#L186

and it compiles now

**Comment by sadr0b0t** [comment]
Internal self-tests are failed anyway

~~~bash
./testCppTestLite > testCppTestLite-gcc-out.txt
~~~

self-test output: http://pastebin.com/uCpCj6Ek

**Comment by doe300** [comment]
GCC 4.8 has a bug in variadic captures for lambdas, see [#41933](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=41933), this bug is fixed in 4.9. CLang has a similar issue, that is why I included the special cases for CLang. 
Thanks for the report, I will add a check, so for older GCC versions, the special cases are used too.

The failing tests fail on purpose.
If you look at the failures, you can see for example, that (in line 14) the assertion '0 == 1' fails, which it should. So, yeah the self-tests fail, but it is on purpose to test the handling and printing of error-messages.
